### PR TITLE
feat(rerank): declarative feature selection with provenance-aware circularity arms

### PR DESCRIPTION
**Pull request by enriquea**
Adds a declarative feature-selection stage to `hvantk rerank`, with provenance-aware
circularity control, plus the dbNSFP typing fix that makes a wide predictor axis usable.

Validated on real data: four phenotypes across three cohorts.

## Why

Feature columns were hand-picked. The rerank-v2 run used **2 of dbNSFP's ~45 predictors**,
a choice inherited from an older spec and never justified. Defending each column in prose
does not scale to a second cohort and is not reproducible. Statistical filtering alone
cannot fix it either: it cannot detect circularity, it *rewards* it — REVEL correlates
with a ClinGen-derived label partly because it was trained on genes like those.

## What this adds

**1. Within-axis feature selection** (`algorithms/rerank/selection.py`)
Univariate AUC screen with within-axis BH-FDR → Spearman redundancy filter → optional
RFECV. Within-axis is what keeps a per-axis ΔAUC meaningful: each axis's best subset is
compared against the baseline's best subset, so a wide axis is not penalised for carrying
redundant columns.

**2. Selection is nested inside cross-validation** (`reranker.py`, `evaluator.py`)
Re-fitted in every fold, on the training slice only, in both the headline and the ablation
paths. `test_selection_nesting.py` is the guardrail and must not be relaxed: on pure noise,
selecting once globally yields **AUC 0.583**; nested, **0.496**.

**3. Provenance arms** (`provenance.py`, `rerank_arms()`)
Every run scores twice over identical folds — `clean` (no provenance conflict against this
label source) and `all`. `clean` is always the headline; `all` exists only so the
circularity channel is a measured number. Circularity is a property of the feature/label
*pair*, so provenance is declared on both sides and the conflict computed.

**4. Per-predictor training provenance in the plugin contract**
`plugin.yaml` datasets may declare `scores: {<column>: {trained_on: [...]}}`, validated by
`manifest.schema.json`. dbNSFP declares it for **55 of 57** rankscores. Omitted = unknown =
usable but never clean. `MutFormer` and `PHACTboost` are deliberately left undeclared —
their training sets could not be confirmed, and a wrong "clean" claim is worse than an
honest unknown.

**5. dbNSFP rankscore typing** (`skills/dbnsfp/builder.py`)
All 57 `*_rankscore` fields now parse to `float64` instead of `str`. Rankscores are the
only dbNSFP surface where predictors are mutually comparable — rank-normalised to 0–1 AND
direction-normalised (the `_converted_` scores are inverted at source, so higher = more
damaging for **every** rankscore). A wide axis built on raw scores would be silently
backwards for SIFT and FATHMM. `"."` maps to missing, never `0.0`.

## Results

`missense_severity` (all 57 rankscores × [mean, frac_gt_0.9]) add-one ΔAUC over constraint,
**clean arm** — every HGMD/ClinVar-trained predictor removed:

| config | n_pos | clean ΔAUC [CI] | all ΔAUC | circularity |
|---|---|---|---|---|
| chd | 266 | **+0.045** [+0.016, +0.074] | +0.083 | +0.038 |
| chd_ndd | 567 | **+0.030** [+0.021, +0.040] | +0.025 | −0.005 |
| epilepsy_dee | 219 | **+0.082** [+0.054, +0.109] | +0.101 | +0.019 |
| schema_scz | 736 | **+0.029** [+0.020, +0.039] | +0.035 | +0.006 |

All four clean CIs exclude zero. On the CHD cohort matrix, like-for-like: baseline 0.7990
→ 5 hand-picked columns 0.8040 → all 57 predictors 0.8670, clean arm 0.8350. Exposing
everything and letting the filters choose beats hand-picking by **+0.031 AUC even on the
honest arm**.

The arms earned their place: they halved CHD's apparent contribution (+0.083 → +0.045),
and the channel is label-specific — same predictors, same matrix, four labels, four
answers (+0.036 / −0.005 / +0.011 / +0.006). A static per-feature leakage flag cannot
produce that. CHD's channel replicated across universes: +0.036 on the 19,448-gene spine
vs +0.031 on the 1,362-gene cohort matrix.

## Two defaults set by measurement, not principle

**RFECV is OFF by default.** Across 4 cohorts / 21 axes it eliminated columns on 4 of the
11 axes wide enough to run it, and **3 of those 4 were the cohort with the fewest positives
(54)** — aggression tracking label scarcity is how inner-CV overfitting looks. It also
pruned the *ablation baseline* axis to a single column on that cohort, which silently
inflates every other axis's ΔAUC. Kept and reachable (`wrapper="rfecv"`), so the
`scikit-learn ^1.4` floor stays (RandomForest needs it for NaN tolerance).

**`auc_global − auc_nested` is documented as a stability diagnostic, not a bias estimate.**
It flips sign on real data: +0.0098 on `all`, −0.0100 on `clean`, same cohort and policy.
It conflates label exposure with committing every fold to one feature set.

## Compatibility

`Config.selection = None` (the default) reproduces the previous code path exactly. The
`hvantk rerank` CLI is unchanged — `config.yaml` has no `selection:` key and `build_config`
never sets one, so this is Python-API only for now.

## Also in this branch

Predating the selection work, from the same run: `insider:interfaces` gene-level dataset,
HGNC pipe-field unquoting, `annotate` gene_id-keyed aggregates + collapse policy, an
`identity` reducer for non-dbNSFP aggregate sources, `count_name` so aggregate axes don't
collide, and a fix for `init_hail` silently discarding kwargs in a fresh process.

## Testing

`1369 passed, 4 skipped, 0 failed` (full suite, Java 11 — without it four Hail tests error
with `UnsupportedClassVersionError`, an environment artifact). dbNSFP hail tests and the
regenerated schema/row snapshots verified on a compute node; the snapshot diff was checked
to be **rankscore-only** (0 unrelated changes in either snapshot file).

## Review notes

- Three files carry the most weight: `selection.py` (the filters), `engine.py` (arms +
  the nesting wiring), and `manifest.schema.json` (the contract change every provider
  inherits).
- `test_selection_nesting.py` encodes the leakage guarantee — worth reading before the
  implementation.
- Known caveats, stated rather than hidden: `EVE` (15% gene coverage) and `LINSIGHT` (12%)
  are coverage-thin; EVE reached the `all` arm and should be treated as suspect until its
  coverage is checked. Four `frac_gt_0.9` columns are constant by construction on tied
  rankscore distributions and are dropped automatically by the univariate filter.

### CODE REVIEW

File: hvantk/algorithms/rerank/engine.py [hunk 1]
```diff
@@ -139,4 +232,79 @@ def _source(col):
...
+    ``n_conflicted`` and ``n_unknown`` separate so the two are not confused.
+    """
+    from hvantk.algorithms.rerank.provenance import DEFAULT_EQUIVALENCE, resolve_arms
+
+    if config.selection is None or config.feature_provenance is None:
+        return {"all": rerank(config)}
+
+    assignment = resolve_arms(
+        config.feature_provenance, config.label_provenance, DEFAULT_EQUIVALENCE
+    )
```
[the remark this answers is not in the document]
Comment (enriquea) [reply hvantk/algorithms/rerank/engine.py:308]: Correct, and worth naming the shape: this is the same failure as 6796d7c3 on this branch ("init_hail silently discarded its kwargs"). `load_policy` returned an equivalence map that had nowhere to live, so the override was dropped and the clean/all split fell back to the defaults — a wrong headline rather than an error.

Fixed in 14a98602: `Config.provenance_equivalence` carries it and `rerank_arms` uses `config.provenance_equivalence or DEFAULT_EQUIVALENCE`. Added `test_custom_equivalence_map_reaches_the_arm_split`, which pins it end-to-end — same provenance and label, default vocabulary gives 0 conflicted columns, a map merging the two sources gives 1.

Your comment also led to an adjacent bug: `_classes()` built its lookup as a dict comprehension, so a source declared under two classes silently last-won by iteration order. That is now rejected, matching the rule applied to a repeated `--prepared` axis in 4af3ac9e.

File: hvantk/skills/dbnsfp/SKILL.md [hunk 2]
```diff
@@ -32,6 +32,23 @@ Supported `params`: `reference_genome` (default `GRCh38`),
 `parse_transcript_scores` (default true), `group_prefixes` (list of str),
 `auto_convert_bgz` (default false).
 
+## Training provenance (`scores:`)
+
+`plugin.yaml` declares `trained_on` for 47 of dbNSFP's predictors — which curated
+database, simulated allele set, or population resource each one was fit on. dbNSFP is
```
[the remark this answers is not in the document]
Comment (enriquea) [reply hvantk/skills/dbnsfp/SKILL.md]: Confirmed — the manifest declares 55, and a2771a2c added 9 declarations without updating the prose.

Fixed in 14a98602. The same stale number appeared in two more places you did not flag: `CHANGELOG.md` also said 47, and the `scores:` block header in `plugin.yaml` itself said "nobody re-reviews 45 scores per analysis". All three now read consistently against the manifest — 55 of 57 rankscores declared, the 2 undeclared being MutFormer and PHACTboost, whose training sets I could not confirm and which are therefore left as unknown rather than claimed clean.
